mu/logic.py

```diff
@@ -855,6 +856,72 @@ def connect_to_status_bar(self, status_bar):
             for mode in self.modes.values():
                 device_selector.device_changed.connect(mode.device_changed)
 
+    def detect_theme_preference(self):
+        """Detect OS dark/light theme preference.
+        Currently implemented for Mac, Windows, GNOME. For any other
+        operating systems, it will always return "day"
+        """
+        # Implementations inspired by the following Stackoverflow submissions:
+        # https://stackoverflow.com/questions/65294987/detect-os-dark-mode-in-python
+
+        def macos_prefers_dark_theme(self):
+            if sys.platform != "darwin":
+                return False
+            cmd = "defaults read -g AppleInterfaceStyle"
+            result = subprocess.run(cmd.split())
+            return result.return_code == 0
+
+        def windows_prefers_dark_theme(self):
+            if sys.platform != "win32":
+                return False
+            try:
+                import winreg
+            except ImportError:
+                return False
+            registry = winreg.ConnectRegistry(None, winreg.HKEY_CURRENT_USER)
+            reg_keypath = (
+                r"SOFTWARE\Microsoft\Windows\CurrentVersion\Themes\Personalize"
+            )
+            try:
+                reg_key = winreg.OpenKey(registry, reg_keypath)
+            except FileNotFoundError:
+                return False
+
+            for i in range(1024):
+                try:
+                    value_name, value, _ = winreg.EnumValue(reg_key, i)
+                    if value_name == "AppsUseLightTheme" and value == 0:
+                        return True
+                except OSError:
+                    break
+            return False
+
+        def gnome_prefers_dark_theme(self):
+            if sys.environ.get("XDG_CURRENT_DESKTOP") != "GNOME":
+                return False
+            # Detect dark mode in GNOME
+            cmd = "gsettings get org.gnome.desktop.interface gtk-theme"
+            result = subprocess.run(cmd.split(), capture_output=True)
+            current_theme = result.stdout.decode("utf-8").strip().strip("'")
+            return current_theme.endswith("-dark")
```

Comment:
There are sooo many ways this can break, the key points being it doesn't consider `gtk-application-prefer-dark-theme` (where the theme name is still Adwaita, though strictly speaking it would be Adwaita:dark) or HighContrastInverse

Side note: *Puts GNOME hat on* We don't currently have dark mode, there is an effort to properly spec one (which would be shared with elementary and hopefully others) but at present everything is unsupported hacks